src/Uno.UWP/ApplicationModel/DataTransfer/DataPackage.macOS.cs

```diff
@@ -0,0 +1,187 @@
+﻿#if __MACOS__
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using AppKit;
+using Foundation;
+using Windows.UI.Core;
+
+namespace Windows.ApplicationModel.DataTransfer
+{
+	public partial class DataPackage
+	{
+		/* UWP has the StandardDataFormats class with properties to define certain types.
+		 * These types are mapped to macOS/iOS as shown in the below table.
+		 * The latest convention in macOS/iOS is to use Uniform Type Identifiers. 
+		 *   https://developer.apple.com/documentation/uniformtypeidentifiers
+		 * However, this is still in beta and only available in macOS 11+.
+		 * NSPasteboard types are therefore used in code until such time that UTI framework
+		 * is available everywhere.
+		 * 
+		 *  UWP                    Uniform           NSPasteboard types                      NSPasteboard types     
+		 *                         Type Identifiers  (currently used)                        (legacy, deprecated)
+		 *  ---------------------  ----------------  --------------------------------------  -------------------------
+		 *  ApplicationLink        UTType.URL        NSPasteboard.NSPasteboardTypeUrl        NSPasteboard.NSUrlType
+		 *  Bitmap                 UTType.Image      NSPasteboard.NSPasteboardTypeTIFF       NSPasteboard.NSTiffType
+		 *                                           NSPasteboard.NSPasteboardTypePNG        -
+		 *  Html                   UTType.HTML       NSPasteboard.NSPasteboardTypeHTML       NSPasteboard.NSHtmlType
+		 *  Rtf                    UTType.RTF        NSPasteboard.NSPasteboardTypeRTF        NSPasteboard.NSRtfType
+		 *                                           NSPasteboard.NSPasteboardTypeRTFD (*1)  NSPasteboard.NSRtfdType
+		 *  StorageItems           UTType.FileURL    NSPasteboard.NSPasteboardTypeFileUrl    -
+		 *  Text                   UTType.PlainText  NSPasteboard.NSPasteboardTypeString     NSPasteboard.NSStringType
+		 *  Uri                    UTType.URL        NSPasteboard.NSPasteboardTypeUrl        NSPasteboard.NSUrlType                                                
+		 *  UserActivityJsonArray  -                 -                                       -
+		 *  WebLink                UTType.URL        NSPasteboard.NSPasteboardTypeUrl        NSPasteboard.NSUrlType 
+		 *  
+		 *  *1 : macOS 10.6 and later
+		 *  
+		 */
+
+		// An Uno-internal abstraction is used to simplify native drag/drop & clipboard integration
+
+		/// <summary>
+		/// Sets the contents of the <see cref="DataPackage"/> to the native clipboard.
+		/// Any conversion between supported standard types is done automatically.
+		/// </summary>
+		/// <param name="content">The contents to set to the native cipboard.</param>
+		internal static void SetToNativeClipboard(DataPackage content)
```

Comment:
If in the partial class, I would have make this method non static

Reply:
The only downside is GetFromNativeClipboard() should be static. As these methods are strongly related it might make sense for other developers to use them the same way. Let's see what the final API in this area looks like after drag/drop is supported, I can certainly change it if you think best.